cmd/lncli/neutrino_active.go

```diff
@@ -0,0 +1,333 @@
+//go:build neutrinorpc
+// +build neutrinorpc
+
+package main
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/lightningnetwork/lnd/lnrpc/neutrinorpc"
+	"github.com/lightningnetwork/lnd/tor"
+	"github.com/urfave/cli"
+)
+
+func getNeutrinoKitClient(ctx *cli.Context) (neutrinorpc.NeutrinoKitClient, func()) {
+	conn := getClientConn(ctx, false)
+
+	cleanUp := func() {
+		conn.Close()
+	}
+
+	return neutrinorpc.NewNeutrinoKitClient(conn), cleanUp
+}
+
+var getNeutrinoStatusCommand = cli.Command{
+	Name:     "status",
+	Usage:    "Returns the status of the running neutrino instance.",
+	Category: "Neutrino",
+	Description: "Returns the status of the light client neutrino " +
+		"instance, along with height and hash of the best block, and " +
+		"a list of connected peers.",
+	Action: actionDecorator(getNeutrinoStatus),
+}
+
+func getNeutrinoStatus(ctx *cli.Context) error {
+	ctxc := getContext()
+	client, cleanUp := getNeutrinoKitClient(ctx)
+	defer cleanUp()
+
+	req := &neutrinorpc.StatusRequest{}
+
+	resp, err := client.Status(ctxc, req)
+	if err != nil {
+		return err
+	}
+
+	printRespJSON(resp)
+
+	return nil
+}
+
+var addPeerCommand = cli.Command{
+	Name:     "addpeer",
+	Usage:    "Add a peer.",
+	Category: "Neutrino",
+	Description: "Adds a new peer that has already been connected to the " +
+		"server.",
+	ArgsUsage: "address",
+	Action:    actionDecorator(addNeutrinoPeer),
+}
+
+func addNeutrinoPeer(ctx *cli.Context) error {
+	ctxc := getContext()
+
+	// Display the command's help message if we do not have the expected
+	// number of arguments/flags.
+	if ctx.NArg() != 1 || ctx.NumFlags() > 0 {
+		return cli.ShowCommandHelp(ctx, "addpeer")
+	}
+
+	client, cleanUp := getNeutrinoKitClient(ctx)
+	defer cleanUp()
+
+	// Check if valid address.
+	ips, err := resolveName(ctx.Args().First())
```

Comment:
Hmm, I think we should instead allow the host itself to be passed. Otherwise, given that `lncli` isn't set up w/ SOCKs support rn, we'd be leaking DNS queries for any pure tor users out there.

Reply:
Changed it.